RestAPI/Models/Repositories/ServiceRepository.cs

```diff
@@ -54,7 +55,32 @@ public async Task SaveUserAsync(User user)
 
         public async Task UpdateBusinessAsync(Business business)
         {
-            //TODO Update
+            Business dbBusiness = await context.Business.FirstOrDefaultAsync(b => b.BusinessID == business.BusinessID);
+            dbBusiness.Update(business);
+            foreach(var w in business.Workdays)
+            {
+                if (w.TimeSheetID == 0) context.TimeSheets.Add(w);
+                else if(w.TimeSheetID > 0)
+                {
+                    TimeSheet dbWorkday = await context.TimeSheets.FirstOrDefaultAsync(workday => workday.TimeSheetID == w.TimeSheetID);
+                    if(dbWorkday != null)
+                    {
+                        dbWorkday.Update(w);
+                    }
+                }
+            }
+            foreach(var p in business.Products)
+            {
+                if (p.ProductID == 0) context.Products.Add(p);
```

Comment:
Maybe it would be better to put context.TimeSheets.Add(w) into curly brackets? It would be more readable.

Reply:
OK, I done that.